### ISSUE TITLE
Error: 400-InvalidParameter, creating workload_osms_dg_policy

### ISSUE THREAD

**Issue by jmalbarranORA** [oracle-quickstart/oci-landing-zones#99]
Hi,

Trying to deploy a landing zone with workload

It fails creating the osms policy

This is the plan:
---
```
  # module.prod_environment.module.workload.module.workload_osms_dg_policy.oci_identity_policy.policy will be created
  + resource "oci_identity_policy" "policy" {
      + ETag           = (known after apply)
      + compartment_id = "ocid1.compartment.oc1..aaaaaaaaxrseqppa3pby7nqukcvb3dh2odw5ko7z7voojfcykmulvdhx7nta"
      + defined_tags   = (known after apply)
      + description    = "Workload OCI Landing Zone OS Management Service Dynamic Group Policy"
      + freeform_tags  = (known after apply)
      + id             = (known after apply)
      + inactive_state = (known after apply)
      + lastUpdateETag = (known after apply)
      + name           = "OCI-P-ELZ-Workload1-OSMS-DG-Policy"
      + policyHash     = (known after apply)
      + state          = (known after apply)
      + statements     = [
          + "Allow dynamic-group OCI-P-ELZ-Workload1-DG to read instance-family in compartment ocid1.compartment.oc1..aaaaaaaaxrseqppa3pby7nqukcvb3dh2odw5ko7z7voojfcykmulvdhx7nta",
          + "Allow dynamic-group OCI-P-ELZ-Workload1-DG to use osms-managed-instances in compartment ocid1.compartment.oc1..aaaaaaaaxrseqppa3pby7nqukcvb3dh2odw5ko7z7voojfcykmulvdhx7nta",
        ]
      + time_created   = (known after apply)
      + version_date   = (known after apply)
    }
```
---

This is the error
---
```
Error: 400-InvalidParameter, Compartment {OCI-ELZ-P-WRK-ADB} does not exist or is not part of the policy compartment subtree
Suggestion: Please update the parameter(s) in the Terraform config as per error message Compartment {OCI-ELZ-P-WRK-ADB} does not exist or is not part of the policy compartment subtree
Documentation: https://registry.terraform.io/providers/oracle/oci/latest/docs/resources/identity_policy 
API Reference: https://docs.oracle.com/iaas/api/#/en/identity/20160918/Policy/UpdatePolicy 
Request Target: PUT https://identity.eu-madrid-1.oci.oraclecloud.com/20160918/policies/ocid1.policy.oc1..aaaaaaaappiw3zve4kg2bcvi7rghxvhnw5wsok6wgvgtq4gz7bpvcz62b4ba 
Provider version: 5.1.0, released on 2023-06-13. This provider is 11 Update(s) behind to current. 
Service: Identity Policy 
Operation Name: UpdatePolicy 
OPC request ID: 885becd12e5fd6dc6dbfaed2e5852b0e/1DB9660B1B9B44F63D87E550A89469B4/D43110D49C0AEF98E64325C88C21A115 


  with module.prod_environment.module.identity.module.security_admin_policy.oci_identity_policy.policy,
  on ../../modules/policies/main.tf line 9, in resource "oci_identity_policy" "policy":
   9: resource "oci_identity_policy" "policy" {
```
---

It fails in both environment P and N (I have included only P for simplicity)

I have tried the same using console, getting same error.

![image](https://github.com/oracle-quickstart/oci-landing-zones/assets/134363018/008a3278-dff9-409a-9234-8ebfba90039a)

It seems that you cannot add a policy referencing the same comparment where is included

**Comment by Reply-emgargano** [comment]
Duplicated by #100

**Comment by VinayKumar611** [comment]
Fixed the issue on latest release v2.2.0.
